services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProvider.java

```diff
@@ -541,6 +541,10 @@ protected JsonWebToken validateToken(String encodedToken, boolean ignoreAudience
         if (!ignoreAudience && !token.hasAudience(getConfig().getClientId())) {
             throw new IdentityBrokerException("Wrong audience from token.");
         }
+        
+        if (!ignoreAudience && !getConfig().getClientId().equals(token.getIssuedFor())) {
```

Comment:
Since azp is optional I would imagine some IdPs don't set it. So this should only check if token.getIssuedFor is not empty.

Reply:
Good point.